src/core/net/netif.hpp

```diff
@@ -214,13 +214,15 @@ class Netif : public InstanceLocator, public LinkedListEntry<Netif>, private Non
     public:
         /**
          * This constructor initializes an `ExternalMulticastAddressIterator` instance to start from the first external
-         * multicast address.
+         * multicast address that matches a given Ip6 address type filter.
          *
-         * @param[in] aNetif  A reference to the Netif instance.
+         * @param[in] aNetif   A reference to the Netif instance.
+         * @param[in] aFilter  The Ip6 address type filter.
          *
          */
-        explicit ExternalMulticastAddressIterator(Netif &aNetif)
+        explicit ExternalMulticastAddressIterator(Netif &aNetif, Address::TypeFilter aFilter)
```

Comment:
How about adding a default for `aFilter` as `aFilter = Ip6::Address::kTypeMulticast`?

Reply:
`IterateExternalMulticastAddresses` already has default value for filter. 

Since we hardly creates `ExternalMulticastAddressIterator` by hand, it not useful to add a default  value for `aFilter`?